commands/src/main/java/org/wildfly/extras/creaper/commands/messaging/AbstractTransportConfigAddCommand.java

```diff
@@ -0,0 +1,230 @@
+package org.wildfly.extras.creaper.commands.messaging;
+
+import org.wildfly.extras.creaper.core.CommandFailedException;
+import org.wildfly.extras.creaper.core.online.OnlineCommand;
+import org.wildfly.extras.creaper.core.online.OnlineCommandContext;
+import org.wildfly.extras.creaper.core.online.OnlineManagementClient;
+import org.wildfly.extras.creaper.core.online.operations.Address;
+import org.wildfly.extras.creaper.core.online.operations.OperationException;
+import org.wildfly.extras.creaper.core.online.operations.Operations;
+import org.wildfly.extras.creaper.core.online.operations.Values;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * An abstract class which implements common properties of connector and acceptor in messaging connections.
+ */
+abstract class AbstractTransportConfigAddCommand implements OnlineCommand {
+
+    protected TransportConfigType type;
+    protected String name;
+    protected String serverName;
+    protected Map<String, String> params;
+    protected boolean replaceExisting;
+
+    //in-vm
+    protected Integer serverId;
+
+    //generic and remote
+    protected String socketBinding;
+
+    //remote
+    protected String factoryClass;
+
+    protected AbstractTransportConfigAddCommand(InVmBuilder builder) {
+        this.type = TransportConfigType.IN_VM;
+        this.serverId = builder.serverId;
+        initCommonOptions(builder);
+    }
+
+    protected AbstractTransportConfigAddCommand(GenericBuilder builder) {
+        this.type = TransportConfigType.GENERIC;
+        this.factoryClass = builder.factoryClass;
+        this.socketBinding = builder.socketBinding;
+        initCommonOptions(builder);
+    }
+
+    protected AbstractTransportConfigAddCommand(RemoteBuilder builder) {
+        this.type = TransportConfigType.REMOTE;
+        this.socketBinding = builder.socketBinding;
+        initCommonOptions(builder);
+    }
+
+    private void initCommonOptions(Builder builder) {
+        this.name = builder.name;
+        this.serverName = builder.serverName;
+        this.params = builder.params;
+        this.replaceExisting = builder.replaceExisting;
+    }
+
+    /**
+     * Get type of transport configuration's instance
+     */
+    protected abstract TransportConfigItem getConfigType();
```

Comment:
Instead of an `abstract` method, this should be a `final` field initialized in a constructor. Subclasses can pass the correct value to the superclass constructor just as easily as overriding a method.